test/Microsoft.AspNetCore.JsonPatch.Test/DynamicObjectAdapterTest.cs

```diff
@@ -0,0 +1,237 @@
+            var segment = "NewProperty";
+            var resolver = new DefaultContractResolver();
+
+            // Act 1
+            var addStatus = adapter.TryAdd(target, segment, resolver, value, out string errorMessage);
+
+            // Assert 1
+            Assert.True(addStatus);
+            Assert.Null(errorMessage);
+            Assert.Equal(value, target.NewProperty);
+
+            // Act 2
+            var getStatus = adapter.TryGet(target, segment, resolver, out object getValue, out string getErrorMessage);
+
+            // Assert 2
+            Assert.True(getStatus);
+            Assert.Null(getErrorMessage);
+            Assert.Equal(getValue, target.NewProperty);
+        }
+
+        [Fact]
+        public void TryGet_ThrowsPathNotFoundException_ForNonExistingProperty()
+        {
+            // Arrange
+            var adapter = new DynamicObjectAdapter();
+            dynamic target = new DynamicTestObject();
+            var segment = "NewProperty";
+            var resolver = new DefaultContractResolver();
+
+            // Act
+            var getStatus = adapter.TryGet(target, segment, resolver, out object getValue, out string getErrorMessage);
+
+            // Assert
+            Assert.False(getStatus);
+            Assert.Null(getValue);
+            Assert.Equal($"The target location specified by path segment '{segment}' was not found.", getErrorMessage);
+        }
+
+        [Fact]
+        public void TryTraverse_FindsNextTarget()
+        {
+            // Arrange
+            var adapter = new DynamicObjectAdapter();
+            dynamic target = new DynamicTestObject();
+            target.NestedObject = new DynamicTestObject();
+            target.NestedObject.NewProperty = "A";
+            var segment = "NestedObject";
+            var resolver = new DefaultContractResolver();
+
+            // Act
+            var status = adapter.TryTraverse(target, segment, resolver, out object nextTarget, out string errorMessage);
+
+            // Assert
+            Assert.True(status);
+            Assert.Null(errorMessage);
+            Assert.Equal(target.NestedObject, nextTarget);
+        }
+
+        [Fact]
+        public void TryTraverse_ThrowsPathNotFoundException_ForNonExistingProperty()
+        {
+            // Arrange
+            var adapter = new DynamicObjectAdapter();
+            dynamic target = new DynamicTestObject();
+            target.NestedObject = new DynamicTestObject();
+            var segment = "NewProperty";
+            var resolver = new DefaultContractResolver();
+
+            // Act
+            var status = adapter.TryTraverse(target.NestedObject, segment, resolver, out object nextTarget, out string errorMessage);
+
+            // Assert
+            Assert.False(status);
+            Assert.Equal($"The target location specified by path segment '{segment}' was not found.", errorMessage);
+        }
+
+        [Theory]
+        [InlineData(1)]
+        [InlineData("new")]
```

Comment:
One small comment here. 

Resist the urge to **prove** things using multiple values.  One of these cases is as good as any other and there's no reason to have them both.

Save theory tests for cases where you have a complicated algorithm to verify. The more experience I gain, the less often I use them.